org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/AbstractProjectsManagerBasedTest.java

```diff
@@ -90,6 +90,14 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
 					clientRequests.put(name, params);
 				}
 				params.add(args[0]);
+			} else if (args.length == 2 && "notify".equals(method.getName())) {
```

Comment:
Because you added "notify" as a method to JavaLanguageClient, you're not getting the notification "notify(param)", but notify("notify", param); 
Removing that method means you cannot use the reflection-based mechanism for checking invocations.  Either you need to build your own or we need to rewrite this mechanism to be based on Endpoints, not proxies